### ISSUE TITLE
What languages are supported?

### ISSUE THREAD

**Issue by davegi** [1nVitr0/plugin-vscode-blocksort#67]
Specifically, I am looking for C, C++ and especially GO. 
Also, have you considered integrated it into an Outliner?
Thanks.

**Comment by 1nVitr0** [comment]
## TLDR

Yes, both GO and C++ are supported. The extension is language agnostic (no specific languages are supported), as long as they are properly indented most syntaxes should work.

***

I have tried to keep the extension as independent as possible of any specific coding languages. Anything of the possible scenarios *should* work, although many programming languages might have some special syntax that may break the block detection:

```
# python-like / yaml-like (indentation based)
match lang:
  case "js":
    print("JavaScript")
  case "cpp" :
    print "C++"
```

```
// JSON-like
{
  "a": 1,
  "b": 2,
  "c": 3
}
```

```
// xml-like
<root>
  <elem>1<elem/>
  <elem>2<elem/>
  <elem>3<elem/>
<root>
```

```
// cpp-like
class Thing
{
  public:
    int a;
    int b;
    int c;
} 
```

```
// java-like
public class Thing {
  @Override
  public function a() {}

  @Override 
  public function b() {}
{
```

***

This means, there is no supper for any specific programming language. If your code is properly formatted (indented and not missing any brackets), most syntaxes should work. I would consider GO somewhat C++-like, so it should be supported. If you find any issues feel free to report them.

You can also look at through the options, they should be somewhat documented in the README. By extending some of them, you might be able to supper some of the quirks your preferred languages may have. If you do, pull requests or issues with options for specific languages are always welcome.

***

As for an outliner, do you mean a folded overview of the file? If so, I'm not sure that is in the scope of the extension, as VS Code already has an outlier for this. It might also confuse the user, as the blocksort outline might not match the folding of the code file (of I had access to the folding in VA Code it would make the whole extension SOOOOO much easier).

**Comment by 1nVitr0** [comment]
What I noticed though, is that `xml-like` languages do not currently support the `@blocksort` markers. (Neither does pure `JSON`, but nothing I can do about that). I might oush an update to that, though that shouldn't concern `GO` or `C++`

**Comment by 1nVitr0** [comment]
@davegi Do you have some bit of sample code? No brackets around if expressions doesn't sound like a problem, but there might be some other issue.

Yeh, I did notice when reading through the options that they're not that well documented. I am happy to help in any way I can. Looking through them might even help me simplifying them, Regexes tend to become more complicated than they need to be...

**Comment by 1nVitr0** [comment]
And dragging around blocks to sort does sound like a pretty good idea, although it definitely sound like a more involved feature. If I do get around to diving deeper into the topic I'll keep you posted.

**Comment by davegi** [comment]
`GO` code w/comments on Block Sort's behavior:
``` GO
package cmd

// fields don't sort when selected or cursor placed anywhere w/in interface
type IFoo interface {
	b() bool
	a()
	c() IFoo
}
// fields don't sort when selected or cursor placed anywhere w/in struct
type S1 struct {
	y int
	x int
	z int
}

type S2[T any] struct {
	y T
	x T
	z T
}

// funcs (w/generic param!) doesn't sort when all three funcs selected
func (s S2[T]) c() IFoo {
	return S1{}
}

func (s S2[T]) b() bool {
	return true
}

func (s S2[T]) a() {
}

// funcs (w/o generic) sorted when all three funcs selected at some point, and then stopped working
// i.e., can't select and Sort Descending

func (s S1) a() {
}

func (s S1) b() bool {
	return true
}

func (s S1) c() IFoo {
	return S1{}
}
```

**Comment by davegi** [comment]
A for the VS Studio extension I was thinking about WRT reorganizing code : [Documentation | CodeMaid](https://tinyl.io/8CeP).

**Comment by davegi** [comment]
I can try to send a video of my interactions, but I noticed that it fairly consistently will work _once_, e.g., 
```GO
type S1 struct {
	x int
	y int
	z int
}
```
Positioning the cursor on the line w/the `y` field (no selection), `Sort Descending` works, but an immediate `Sort Ascending` fails. Seems similar w/the member functions.

**Comment by 1nVitr0** [comment]
Should be fixed in #68 

If not, let me know

**Comment by davegi** [comment]
Uninstalled, _reloaded_, and re-installed...selecting the original  example members, _i.e._, `a()`, `b()`...and then invoking Sort Block from the _Quick Fix_ menu, still did not sort the functions.

**Comment by 1nVitr0** [comment]
Sorry, I haven't worked on this extension for a while. I will look into it in the following weeks if the issue still persists

**Comment by 1nVitr0** [comment]
@davegi Will close this for now, feel free to open another issue and reference this one if the problem still persists
